pyvrp/read.py

```diff
@@ -100,15 +100,12 @@ def read(
 
     instance = vrplib.read_instance(where, instance_format=instance_format)
 
-    # A priori checks
-    if "dimension" in instance:
-        dimension: int = instance["dimension"]
-    else:
-        if "demand" not in instance:
-            raise ValueError("File should either contain dimension or demands")
-        dimension = len(instance["demand"])
```

Comment:
There wasn't really a need for this dance to get the problem size: if it's available, we get it from `dimension`, and if it's not, we get it from the number of rows in the dist/dur matrices.